constellation-app/constellation
# Feature/table default columns latest

### Prerequisites


    for the "CI / Unit Tests") to complete first. Failed Unit tests can be 
    debugged by adding the label "verbose logging" to the GitHub PR.

### Description of the Change
- To get/set the default columns for new graphs
- Added TableDefaultsColumnsProvider which implement new interface TableDefaultColumns to get default columns
- ColumnVisibilityContext Menu uses lookup of TableDefaultColumns to set the default columns
- To add ability to create other objects of lookup type TableDefaultColumns with lower position order if they wish 



### Alternate Designs



### Why Should This Be In Core?



### Benefits


- To get/set the default columns for new graphs
- To add ability to create other objects of lookup type TableDefaultColumns with lower position order if they wish for different set of default columns

### Possible Drawbacks



### Verification Process

1. Open constellation
2. Open new graph
3. Open table view - observe that default columns are displayed in table view (currently there are default columns which are the same as primary key columns)

And

1. Open an existing graph with multiple nodes and transactions
2. On table view, click on the first button which displays a context menu to show various columns
3. Select Show All Columns (and observe all columns are displayed in table view. Then select Show Default Columns (and observe that the default columns are now displayed in table view, which would be the same as primary key columns)

You could also save the table view with the default columns as before.




### Applicable Issues
#1882
